src/components/page/page_header/page_header.test.tsx

```diff
@@ -124,4 +124,48 @@ describe('OuiPageHeader', () => {
       });
     });
   });
+
+  describe('deprecation', () => {
+    it('should console 1 deprecation warning without repetition', () => {
+      console.warn = jest.fn();
+
+      const component = mount(<OuiPageHeader iconType="dashboardApp" />);
+      component.setProps({ iconType: 'database' });
+
+      expect(console.warn).toHaveBeenCalledTimes(1);
+      expect(console.warn).toHaveBeenCalledWith(
+        '[DEPRECATED] The `iconType` prop is deprecated and will be removed in v2.0.0'
+      );
```

Comment:
Really like that we're testing to make sure only 1 message is logged even when props are changed